Bottom: d61714d580d97da9f836e7b65c37a4d61a205a8b
Top:    d1e67fef1b22263e0d41bddeb746d8bc6470d487
Author: Peter Clifton <pcjc2@cam.ac.uk>
Date:   2009-02-08 02:04:30 +0000

Workaround crash where contour is passed as NULL to poly_PreContour












---

diff --git a/src/polygon.c b/src/polygon.c
index c0aa76d..181a6c3 100644
--- a/src/polygon.c
+++ b/src/polygon.c
@@ -247,6 +247,16 @@ original_poly (PolygonType * p)
       }
   }
   END_LOOP;
+  if (contour == NULL)
+    {
+      printf ("How did that escape - did the loop iterate zero times??\n");
+      POLYGONPOINT_LOOP (p);
+        {
+          printf ("Hello\n");
+        }
+      END_LOOP;
+      return NULL;
+    }
   poly_PreContour (contour, TRUE);
   /* make sure it is a positive contour */
   if ((contour->Flags.orient) != PLF_DIR)
